### PR TITLE
Update render stats and scene export

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

**Comment by jdent02** [comment]
#160

**Comment by jdent02** [comment]
#149

File: export.py [hunk 1]
```diff
@@ -78,6 +78,39 @@ def execute(self, context):
...
+        return renderer.engine == 'APPLESEED_RENDER'
+
+    def execute(self, context):
+        scene = context.scene
+        frame_start = scene.frame_start
+        frame_end = scene.frame_end
+
+        for frame in range(frame_start, frame_end + 1):
+            scene.frame_set(frame)
+            print(scene.frame_current)
```
Comment (dictoon) [inline export.py]: That looks like a debug output, remove?

File: export.py [hunk 2]
```diff
@@ -78,6 +78,39 @@ def execute(self, context):
...
+            file_split = self.filepath.split(".")
+            export_path = util.realpath(os.path.join(file_split[0] + '_{0}.{1}'.format(frame, file_split[1])))
+            print(export_path)
```
Comment (dictoon) [inline export.py]: Same question here.

File: export.py [hunk 3]
```diff
@@ -78,6 +78,39 @@ def execute(self, context):
...
+    def execute(self, context):
+        scene = context.scene
+        frame_start = scene.frame_start
+        frame_end = scene.frame_end
+
+        for frame in range(frame_start, frame_end + 1):
+            scene.frame_set(frame)
+            print(scene.frame_current)
+            file_split = self.filepath.split(".")
+            export_path = util.realpath(os.path.join(file_split[0] + '_{0}.{1}'.format(frame, file_split[1])))
```
Comment (dictoon) [inline export.py]: Use `{0}_{1}.{2}` to avoid string concatenation.

File: projectwriter.py [hunk 4]
```diff
@@ -76,6 +76,9 @@ def write(self, scene, file_path):
         # Root path of the exported project.
         self._root_path = os.path.dirname(util.realpath(file_path))
 
+        # Whether this is an exported animation
```
Comment (dictoon) [inline projectwriter.py]: End sentence with `.`

File: projectwriter.py [hunk 5]
```diff
@@ -459,7 +463,7 @@ def __emit_mesh_object(self, scene, object, mesh, mesh_faces, mesh_uvtex, new_as
 
         object_name = object.name
 
-        mesh_filename = object_name + ".obj"
+        mesh_filename = object_name + ".obj" if not do_def_blur else object_name + "_{0}.obj".format(scene.frame_current)
```
Comment (dictoon) [inline projectwriter.py]: Use `{0}_{1}`.

File: projectwriter.py [hunk 6]
```diff
@@ -536,9 +540,8 @@ def __emit_def_mesh_object(self, scene, object, mesh, mesh_faces, mesh_uvtex):
         if len(mesh_faces) == 0:
             self.__info("Skipping object '{0}' since it has no faces once converted to a mesh.".format(object.name))
             return []
-
         object_name = object.name
-        mesh_filename = object_name + "_deform.obj"
+        mesh_filename = object_name + "_{0}_deform.obj".format(scene.frame_current)
```
Comment (dictoon) [inline projectwriter.py]: Same here.

File: projectwriter.py [hunk 7]
```diff
@@ -468,7 +472,7 @@ def __emit_mesh_object(self, scene, object, mesh, mesh_faces, mesh_uvtex, new_as
                 os.mkdir(meshes_path)
             if scene.appleseed.export_mode == 'all':
                 export_mesh = True
-            if scene.appleseed.export_mode == 'partial' and not os.path.exists(mesh_filepath):
+            if scene.appleseed.export_mode == 'partial' or self.animation and not os.path.exists(mesh_filepath):
```
Comment (dictoon) [inline projectwriter.py]: The logic isn't obvious here, add parentheses.

File: projectwriter.py [hunk 8]
```diff
@@ -550,7 +553,7 @@ def __emit_def_mesh_object(self, scene, object, mesh, mesh_faces, mesh_uvtex):
...
```
Comment (dictoon) [inline projectwriter.py]: Add parentheses to clarify logic.

File: projectwriter.py [hunk 9]
```diff
@@ -578,7 +581,7 @@ def __emit_def_curves_object(self, scene, object, psys):
                 os.mkdir(meshes_path)
             if scene.appleseed.export_mode == 'all':
                 export_curves = True
-            if scene.appleseed.export_mode == 'partial' and not os.path.exists(curves_filepath):
+            if scene.appleseed.export_mode == 'partial' or self.animation and not os.path.exists(curves_filepath):
```
Comment (dictoon) [inline projectwriter.py]: Add parentheses to clarify logic.

File: render.py [hunk 10]
```diff
@@ -189,8 +194,21 @@ def __render_project_file(self, scene, project_filepath, project_dir=None):
             max_x = width - 1
             max_y = height - 1
 
+        # Compute number of tiles.
+        if (max_y - min_y + 1) % scene.appleseed.tile_height == 0:
```
Comment (dictoon) [inline render.py]: There's almost certainly a nicer way to compute this, possibly using `ceil()`, i.e.

    vertical_tiles = int(ceil((max_y - min_y + 1) / scene.appleseed.tile_height))

File: render.py [hunk 11]
```diff
@@ -189,8 +194,21 @@ def __render_project_file(self, scene, project_filepath, project_dir=None):
...
+            vertical_tiles = int((max_y - min_y + 1) / scene.appleseed.tile_height)
+        else:
+            vertical_tiles = int((max_y - min_y + 1) / scene.appleseed.tile_height) + 1
+        if (max_x - min_x + 1) % scene.appleseed.tile_width == 0:
+            horizontal_tiles = int((max_x - min_x + 1) / scene.appleseed.tile_width)
+        else:
+            horizontal_tiles = int((max_x - min_x + 1) / scene.appleseed.tile_width) + 1
+        self.number_of_tiles = vertical_tiles * horizontal_tiles
```
Comment (dictoon) [inline render.py]: Rename `number_of_tiles` to `total_tiles` for consistency with `total_passes` and `total_pixels`.

File: render.py [hunk 12]
```diff
@@ -359,6 +388,10 @@ def __process_tile_highlight_chunk(self, process, min_x, min_y, max_x, max_y):
         self.__draw_hline(x1 - bracket_width + 1, y0, bracket_width, bracket_color)
         self.__draw_vline(x1, y0, bracket_height, bracket_color)
 
+        if self.initial_start:
```
Comment (dictoon) [inline render.py]: Can we infer `initial_start` from the existing state variables such as `rendered_tiles`, `pass_number`, etc.?

File: render.py [hunk 13]
```diff
@@ -372,3 +405,10 @@ def __draw_vline(self, x, y, length, color):
         layer = result.layers[0].passes[0]
         layer.rect = [color] * length
         self.end_result(result)
+
+    def format_seconds_to_hhmmss(self, seconds):
```
Comment (dictoon) [inline render.py]: You can use the `@staticmethod` attribute here.

File: ui/render.py [hunk 14]
```diff
@@ -254,7 +284,8 @@ def draw(self, context):
 
 
 def register():
-    bpy.types.RENDER_PT_render.COMPAT_ENGINES.add('APPLESEED_RENDER')
+    # bpy.types.RENDER_PT_render.COMPAT_ENGINES.add('APPLESEED_RENDER')
```
Comment (dictoon) [inline ui/render.py]: Why is this commented out?

**Comment by jdent02** [comment]
Should have everything covered.

File: projectwriter.py [hunk 15]
```diff
@@ -427,7 +431,7 @@ def __emit_curves_object(self, scene, object, psys, new_assembly=False):
                 os.mkdir(meshes_path)
             if scene.appleseed.export_mode == 'all':
                 export_curves = True
-            if scene.appleseed.export_mode == 'partial' and not os.path.exists(curves_filepath):
+            if scene.appleseed.export_mode == 'partial' or self._animation and not os.path.exists(curves_filepath):
```
Comment (dictoon) [inline projectwriter.py]: Add parentheses.

File: ui/render.py [hunk 16]
```diff
@@ -40,6 +40,36 @@ def poll(cls, context):
...
+
+        layout.prop(asr_scene_props, "export_only", text="Export appleseed Files Only")
+        row = layout.row(align=True)
+        if asr_scene_props.export_only:
+            row.operator("appleseed.export_scene", text="Export Frame", icon='RENDER_STILL')
+            row.operator("appleseed.export_anim_scene", text="Export Animation", icon='RENDER_ANIMATION')
+        else:
+            row.operator("render.render", text="Render", icon='RENDER_STILL')
+            row.operator("render.render", text="Animation", icon='RENDER_ANIMATION').animation = True
+        row.operator("sound.mixdown", text="Audio", icon='PLAY_AUDIO')
```
Comment (dictoon) [inline ui/render.py]: Audio?

**Pull request merged**